libs/akita-ng-router-store/src/lib/router.service.ts

```diff
@@ -55,69 +52,55 @@ export class RouterService {
     this.routerQuery
       .select(state => state)
       .subscribe(s => {
-        this.routerState = s;
+        this.lastRouterState = s;
         this.navigateIfNeeded();
       });
   }
 
-  private shouldDispatchRouterNavigation(): boolean {
-    if (!this.routerState) return true;
-    return !this.navigationTriggeredByDispatch;
-  }
-
   private navigateIfNeeded(): void {
-    if (!this.routerState || !this.routerState.state) {
+    if (!this.lastRouterState || !this.lastRouterState.state || this.dispatchTriggeredByRouter) {
       return;
     }
-    if (this.dispatchTriggeredByRouter) return;
 
-    if (this.router.url !== this.routerState.state.url) {
+    if (this.router.url !== this.lastRouterState.state.url) {
       this.navigationTriggeredByDispatch = true;
       setSkipAction();
-      this.router.navigateByUrl(this.routerState.state.url);
+      this.router.navigateByUrl(this.lastRouterState.state.url);
     }
   }
 
   private setUpStateRollbackEvents(): void {
     this.router.events.subscribe(e => {
-      if (e instanceof RoutesRecognized) {
-        this.lastRoutesRecognized = e;
-      } else if (e instanceof ResolveEnd) {
-        this.resolveEnd(e);
+      if (e instanceof RoutesRecognized || e instanceof GuardsCheckEnd || e instanceof ResolveEnd) {
```

Comment:
```suggestion
      if ([RoutesRecognized, GuardsCheckEnd, ResolveEnd].some(event => e instanceof event)) {
```

Reply:
That syntax doesn't work with type narrowing :(